doc/hotspot-style.md

```diff
@@ -770,6 +770,23 @@ ordering, which may differ from (may be stronger than) sequentially
 consistent.  There are algorithms in HotSpot that are believed to rely
 on that ordering.
 
+### Initializing variables with static storage duration
+
+Avoid variables with static storage duration and non-constant initialization,
+or with non-trivial destruction.  Such variables can lead to the so-called
```

Comment:
I wonder about the phrase "non-constant". There are a few places where we set up static objects that are non-constant, but doesn't depend on anything. For example, `ZGC_ONLY(static ZArguments          zArguments;)
`. Is `non-constant` precise enough to describe what we want to prevent?